sync-android/src/main/java/com/cloudant/sync/sqlite/android/encryption/EncryptionUtils.java

```diff
@@ -0,0 +1,298 @@
+          
+          if (!LOADED_LIBS.contains(library)) {
+
+               // we only support "armeabi" and "x86"
+               final String ARMEABI = "armeabi";
+               final String X86 = "x86";
+
+               String arch = System.getProperty("os.arch");  // the architecture we're running on
+               String nonArch = null;  // the architecture we are NOT on
+               if (arch != null && arch.toLowerCase().startsWith("i")) {  // i686
+                    arch = X86;
+                    nonArch = ARMEABI;
+               } else {
+                    arch = ARMEABI;
+                    nonArch = X86;
+               }
+
+               final String libPath = "featurelibs" + File.separator +  arch + File.separator + library;
+               File sourceLocation = new File(context.getFilesDir(), libPath + ".zip");
+
+               // recursively delete the architecture folder that will never be used:
+               File nonArchStorage = new File(context.getFilesDir(), "featurelibs" + File.separator + nonArch);
+               deleteDirectory(nonArchStorage);
+
+               File targetFile = new File(context.getFilesDir(), library);
+
+               // delete the target
+               targetFile.delete();
+
+               Log.d(TAG,"Extracting zip file: " + libPath);
+               try{
+                    InputStream istr = context.getAssets().open(libPath + ".zip");
+                    unpack(istr, targetFile.getParentFile());
+               }
+               catch(IOException e){
+            	   e.printStackTrace(); 
+            	   Log.d(TAG,"Error extracting zip file: " + e.getMessage());
+               }
+
+               Log.d(TAG,"Loading library using System.load: " + targetFile.getAbsolutePath());
+
+               // delete the original zip, which is now extracted:
+               //sourceLocation.delete();
+
+               System.load(targetFile.getAbsolutePath());
+               
+               LOADED_LIBS.add (library);
+
+               //Load context in security manager
+               SecurityManager.getInstance(context);
+          }
+     }
+
+     /**
+      * Delete a file or directory, including all its children.  The method name "deleteDirectory"
+      * is retained for legacy callers, but can take a directory or file as a parameter.
+      *
+      * @param fileOrDirectory The {@link File} object represents the directory to delete.
+      * @return true if the directory was deleted, false otherwise.
+      */
+     public static boolean deleteDirectory(File fileOrDirectory) {
+          if (fileOrDirectory.isDirectory()) {
+               for (File child : fileOrDirectory.listFiles()) {
+                    deleteDirectory(child);
+               }
+          }
+          return fileOrDirectory.delete();
+     }
+
+
+     public static void unpack(InputStream in, File targetDir) throws IOException {
```

Comment:
Yea, this code was pulled from a JSONStore class.  Many of these methods were not needed but I had not cleaned up the class yet.